packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -19,30 +19,76 @@ import {
     Modify,
     MoveId,
     MoveIn,
+    NewAttach,
     MoveOut,
     ObjectMark,
+    InputSpanningMark,
     Reattach,
     ReturnFrom,
     ReturnTo,
-    SizedMark,
-    SizedObjectMark,
     Skip,
+    Muted,
+    Mutable,
+    OutputSpanningMark,
+    Changeset,
 } from "./format";
+import { MarkListFactory } from "./markListFactory";
 
 export function isModify<TNodeChange>(mark: Mark<TNodeChange>): mark is Modify<TNodeChange> {
     return isObjMark(mark) && mark.type === "Modify";
 }
 
+export function isNewAttach<TNodeChange>(mark: Mark<TNodeChange>): mark is NewAttach<TNodeChange> {
+    return isObjMark(mark) && (mark.type === "Insert" || mark.type === "MoveIn");
+}
+
 export function isAttach<TNodeChange>(mark: Mark<TNodeChange>): mark is Attach<TNodeChange> {
-    return (
-        (isObjMark(mark) && (mark.type === "Insert" || mark.type === "MoveIn")) || isReattach(mark)
-    );
+    return isNewAttach(mark) || isReattach(mark);
+}
+
+export function isAttachInGap<TNodeChange>(mark: Mark<TNodeChange>): mark is Attach<TNodeChange> {
+    return isNewAttach(mark) || isActiveReattach(mark) || isBlockedReattach(mark);
 }
 
 export function isReattach<TNodeChange>(mark: Mark<TNodeChange>): mark is Reattach<TNodeChange> {
     return isObjMark(mark) && (mark.type === "Revive" || mark.type === "ReturnTo");
 }
 
+export function isActiveReattach<TNodeChange>(
+    mark: Mark<TNodeChange>,
+): mark is Reattach<TNodeChange> & Muted {
+    // No need to check Reattach.lastDeletedBy because it can only be set if the mark is muted
+    return isReattach(mark) && !isMuted(mark);
+}
+
+export function isMutedReattach<TNodeChange>(
+    mark: Mark<TNodeChange>,
+): mark is Reattach<TNodeChange> & Muted {
+    return isReattach(mark) && isMuted(mark);
+}
+
+export function isMutedDetach<TNodeChange>(
+    mark: Mark<TNodeChange>,
+): mark is Detach<TNodeChange> & Muted {
+    return isDetachMark(mark) && isMuted(mark);
+}
+
+export function isSkipLikeReattach<TNodeChange>(
```

Comment:
Yeah, it sounds more like a different flavor of reattach rather than a reattach with a specific status, and it doesn't convey what the status is. The best way I would describe the status is "(a reattach) whose target cells are already populated/occupied", but I don't see how to work that into a nice name.
It may help if we can find a better name for the "Blocked" state too.